src/providers/drafts.ts

```diff
@@ -0,0 +1,118 @@
+import { Language, License, VideoCategory } from '@/api/queries'
+import { createStore } from '@/store'
+import { createId } from '@/utils/createId'
+import { readFromLocalStorage } from '@/utils/localStorage'
+
+export type CommonDraftProps = {
+  id: string
+  channelId: string
+  updatedAt: string
+}
+
+export type Draft = VideoDraft
+
+export type DraftType = 'video'
+
+export type VideoDraft = {
+  type: 'video'
+  title?: string | null
+  description?: string | null
+  isPublic?: boolean | null
+  publishedBeforeJoystream?: string | null
+  hasMarketing?: boolean | null
+  isExplicit?: boolean | null
+  language?: Language['iso'] | null
+  category?: VideoCategory['id'] | null
+  licenseCode?: License['code'] | null
+  licenseCustomText?: License['customText'] | null
+  licenseAttribution?: License['attribution'] | null
+} & CommonDraftProps
+
+export type RawDraft = Omit<Draft, 'id' | 'updatedAt'>
+
+export type UnseenDraft = {
+  draftId: string
+  channelId: string
+}
+
+interface DraftStoreState {
+  allDrafts: Draft[]
+  allUnseenDrafts: UnseenDraft[]
+}
+
+interface DraftStoreActions {
+  addDraft: (draft: RawDraft, explicitId?: string) => Draft
+  updateDraft: (draftId: string, draftProps: RawDraft) => void
+  getDraft: (draftId: string) => Draft | undefined
+  getDraftsForChannel: (draftIds: string) => Draft[]
+  getUnseenDraftsForChannel: (draftIds: string) => UnseenDraft[]
+  removeDrafts: (draftIds: string[]) => void
+  removeAllDrafts: (channelId: string) => void
+  removeAllUnseenDrafts: (channelId: string) => void
+}
+
+export const useDraftStore = createStore<DraftStoreState, DraftStoreActions>(
+  {
+    state: {
+      allDrafts: [], // includes drafts for different channels
+      allUnseenDrafts: [], // includes unseenDrafts for different channels
```

Comment:
I'm wondering, do we need to keep those two separate? Maybe some `seen: boolean` field on draft would do the trick? I remember there was some discussion like this originally, but don't remember what the arguments were

Reply:
Yeah, before it was cumbersome but I don't see why keep them separate now, will merge

Reply:
Great